aiplatform/apiv1beta1/deployment_resource_pool_client.go

```diff
@@ -0,0 +1,672 @@
+		internaloption.EnableJwtWithScope(),
+		option.WithGRPCDialOption(grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(math.MaxInt32))),
+	}
+}
+
+func defaultDeploymentResourcePoolCallOptions() *DeploymentResourcePoolCallOptions {
+	return &DeploymentResourcePoolCallOptions{
+		CreateDeploymentResourcePool: []gax.CallOption{},
+		GetDeploymentResourcePool:    []gax.CallOption{},
+		ListDeploymentResourcePools:  []gax.CallOption{},
+		DeleteDeploymentResourcePool: []gax.CallOption{},
+		QueryDeployedModels:          []gax.CallOption{},
+		GetLocation:                  []gax.CallOption{},
+		ListLocations:                []gax.CallOption{},
+		GetIamPolicy:                 []gax.CallOption{},
+		SetIamPolicy:                 []gax.CallOption{},
+		TestIamPermissions:           []gax.CallOption{},
+		CancelOperation:              []gax.CallOption{},
+		DeleteOperation:              []gax.CallOption{},
+		GetOperation:                 []gax.CallOption{},
+		ListOperations:               []gax.CallOption{},
+		WaitOperation:                []gax.CallOption{},
+	}
+}
+
+// internalDeploymentResourcePoolClient is an interface that defines the methods available from Vertex AI API.
+type internalDeploymentResourcePoolClient interface {
+	Close() error
+	setGoogleClientInfo(...string)
+	Connection() *grpc.ClientConn
+	CreateDeploymentResourcePoolOperation(name string) *CreateDeploymentResourcePoolOperation
+	DeleteDeploymentResourcePoolOperation(name string) *DeleteDeploymentResourcePoolOperation
+	GetLocation(context.Context, *locationpb.GetLocationRequest, ...gax.CallOption) (*locationpb.Location, error)
+	ListLocations(context.Context, *locationpb.ListLocationsRequest, ...gax.CallOption) *LocationIterator
+	GetIamPolicy(context.Context, *iampb.GetIamPolicyRequest, ...gax.CallOption) (*iampb.Policy, error)
+	SetIamPolicy(context.Context, *iampb.SetIamPolicyRequest, ...gax.CallOption) (*iampb.Policy, error)
+	TestIamPermissions(context.Context, *iampb.TestIamPermissionsRequest, ...gax.CallOption) (*iampb.TestIamPermissionsResponse, error)
+	CancelOperation(context.Context, *longrunningpb.CancelOperationRequest, ...gax.CallOption) error
+	DeleteOperation(context.Context, *longrunningpb.DeleteOperationRequest, ...gax.CallOption) error
+	GetOperation(context.Context, *longrunningpb.GetOperationRequest, ...gax.CallOption) (*longrunningpb.Operation, error)
+	ListOperations(context.Context, *longrunningpb.ListOperationsRequest, ...gax.CallOption) *OperationIterator
+	WaitOperation(context.Context, *longrunningpb.WaitOperationRequest, ...gax.CallOption) (*longrunningpb.Operation, error)
+}
+
+// DeploymentResourcePoolClient is a client for interacting with Vertex AI API.
+// Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.
+//
+// A service that manages the DeploymentResourcePool resource.
+type DeploymentResourcePoolClient struct {
```

Comment:
Is there a reason why this client doesn't have a `restClient`? Something to do with only having LRO RPCs?